src/toxcore/dht/server/mod.rs

```diff
@@ -169,6 +170,9 @@ pub struct Server {
     /// Lru cache for precomputed keys. It stores precomputed keys to avoid
     /// redundant calculations.
     precomputed_keys: PrecomputedCache,
+    /// Counters for incoming/outgoing udp packets.
+    /// Counters are in DhtCodec object and can be accessed with.
+    codec: Option<DhtCodec>,
```

Comment:
Do we really need `DhtCodec` inside the server?

Reply:
Well, no. We need a way to get stats from DhtServer to send them back to user in `server.set_bootstrap_info...`.

I wonder how can we do that for tcp relay?

Reply:
`set_bootstrap_info` uses callback function. This function can have access to DhtCodec and to tcp relay.

Reply:
@kurnevsky so what's your possible solution?

Reply:
I'd suggest to pass counters via Codec's constructor. This way we can put these counters inside Arc and use anywhere else we want them. The same approach can be used for tcp relay.